### DISCUSSION TITLE
Is Oceananigans Capable of Performing Tsunami Modelling?

### DISCUSSION THREAD

**Discussion by MGYBY** [CliMA/Oceananigans.jl#1955] [Computational science!] [answered]
Hello all,
I am a total beginner of Oceananigans. As the title suggests, I have the following questions which  I would be  appreciated if you could help me. Is Oceananigans capable of performing tsunami modelling? In particular, is Oceananigans capable of producing the tsunami inundation map and dealing with wetting-drying? Thank you in advance for your answers.
BYY

**Comment by francispoulin** [comment, marked as answer]
Hello @MGYBY 
I am glad you are interested in using Oceananigans.  Unfortunately, at present, the library does not allow for wetting and drying.
Francis

**Reply by MGYBY** [reply]
Thanks for the answer.

**Reply by glwagner** [reply]
I guess this would be something to implement for `ShallowWaterModel` (using dynamically changing bathymetry + immersed boundaries?) It does not sound trivial though.

**Comment by francispoulin** [comment]
None of the advection schemes that are currently supported in Oceananigans are positive preserving, hence one problem why we can't currently allow for wetting and drying.

As a first step in this direction, we should create a positive preserving advection scheme.  There are two options that come to mind,

  1.  There is a way to modify WENO5 to make it positive preserving.
  2.  First order-upwinding is naturally positive preserving.

I will start with the second since it's way easier, even though it is only first order accurate, and that will also be helpful for comparing with the more complicated positive preserving WENO5.


------------------
Francis Poulin
Associate Editor, Canadian Journal of Physics
Professor of Applied Mathematics and Physics and Astronomy
University of Waterloo

email:           ***@***.***
Web:            https://uwaterloo.ca/poulin-research-group/
Telephone:  +1 519 888 4567 x32637
[cid:76187fd2-7da5-4e10-a8a7-f81d6fbf329d]
________________________________
